src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeRecipientRepository.kt

```diff
@@ -24,7 +24,7 @@ interface BarcodeRecipientRepository : JpaRepository<BarcodeRecipient, Long> {
 data class BarcodeRecipient(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
-  val id: Long = -1,
+  val id: Long? = null,
```

Comment:
I don't like either of these solutions! We want to say nullable if not saved but non-nullable if saved. As it's quite rare to be dealing with an entity that's not saved (apart from when saving it) I have in the past leaned towards non-nullable and value of -1.

Reply:
Hmmm, I dont like the -1 option :) But yes, we want it to be nullable only when its a new entity.

The reason I dont like -1 is that when creating a new entity we present it to JPA with an ID of -1, and it's fortuitous that JPA/postgres determines that what we mean is a new entity rather than updating the entity whose ID is -1
That might be because of the db schema where we say the id column is of type `serial` so perhaps -1 is a 'magic' number in that respect? (thats a guess - I dont know for sure)

I wonder if there's a better way of doing this ... I'll have a play ....